File: Task-03/Go/Sub-Task 4/Pattern-FileHandling2.go
package main

import (
    "io/ioutil"
    "strconv"
    "strings"
)

func main() {
    content, _ := ioutil.ReadFile("input.txt")
    n, _ := strconv.Atoi(strings.TrimSpace(string(content)))

    var output string
    for i := 0; i < n; i++ {
        output += strings.Repeat(" ", n-i-1) + strings.Repeat("*", 2*i+1) + "\n"
    }
    for i := n - 2; i >= 0; i-- {
        output += strings.Repeat(" ", n-i-1) + strings.Repeat("*", 2*i+1) + "\n"
    }

    ioutil.WriteFile("output.txt", []byte(output), 0644)
}

File: Task-03/Haskell/Sub-Task 4/Pattern-FileHandling2.hs
main = do
    input <- readFile "input.txt"
    let n = read input :: Int
    let upper = [replicate (n - i - 1) ' ' ++ replicate (2 * i + 1) '*' | i <- [0..n-1]]
    let lower = [replicate (i + 1) ' ' ++ replicate (2 * (n - i - 2) + 1) '*' | i <- [0..n-2]]
    writeFile "output.txt" (unlines (upper ++ lower))
